### PR TITLE
separated services, adding some missing api calls, made others uniform

### PR DESCRIPTION
**Pull request by dvoet**

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/rawls/webservice/EntityApiService.scala [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+  @Path("")
+  @ApiOperation(value = "Create entity in a workspace",
+    nickname = "create entity",
+    httpMethod = "POST",
+    produces = "application/json",
+    response = classOf[Entity])
+  @ApiImplicitParams(Array(
+    new ApiImplicitParam(name = "workspaceNamespace", required = true, dataType = "string", paramType = "path", value = "Workspace Namespace"),
+    new ApiImplicitParam(name = "workspaceName", required = true, dataType = "string", paramType = "path", value = "Workspace Name"),
+    new ApiImplicitParam(name = "entityType", required = true, dataType = "string", paramType = "path", value = "Entity Type"),
```
Comment (jmthibault79) [inline src/main/scala/org/broadinstitute/dsde/rawls/webservice/EntityApiService.scala]: `entityType` and `entityName` are not used here.

File: src/main/scala/org/broadinstitute/dsde/rawls/webservice/EntityApiService.scala [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+    new ApiImplicitParam(name = "workspaceNamespace", required = true, dataType = "string", paramType = "path", value = "Workspace Namespace"),
+    new ApiImplicitParam(name = "workspaceName", required = true, dataType = "string", paramType = "path", value = "Workspace Name"),
+    new ApiImplicitParam(name = "entityType", required = true, dataType = "string", paramType = "path", value = "Entity Type")
+  ))
+  @ApiResponses(Array(
+    new ApiResponse(code = 200, message = "Successful Request"),
+    new ApiResponse(code = 404, message = "Workspace or entityType does not exist"),
+    new ApiResponse(code = 500, message = "Rawls Internal Error")
+  ))
+  def listEntitiesRoute = cookie("iPlanetDirectoryPro") { securityTokenCookie =>
```
Comment (jmthibault79) [inline src/main/scala/org/broadinstitute/dsde/rawls/webservice/EntityApiService.scala]: I would prefer a name like listEntitiesPerTypeRoute here

File: src/main/scala/org/broadinstitute/dsde/rawls/webservice/MethodConfigApiService.scala [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+    listMethodConfigurationsRoute
+
+  @Path("")
+  @ApiOperation(value = "Create Method configuration in a workspace",
+    nickname = "create method configuration",
+    httpMethod = "POST",
+    produces = "application/json",
+    response = classOf[MethodConfiguration])
+  @ApiImplicitParams(Array(
+    new ApiImplicitParam(name = "methodConfigJson", required = true, dataType = "org.broadinstitute.dsde.rawls.model.MethodConfiguration", paramType = "body", value = "Method Configuration contents")
```
Comment (jmthibault79) [inline src/main/scala/org/broadinstitute/dsde/rawls/webservice/MethodConfigApiService.scala]: missing `workspaceNamespace` and `workspaceName`

File: src/main/scala/org/broadinstitute/dsde/rawls/webservice/MethodConfigApiService.scala [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+  @Path("/{configNamespace}/{configName}/rename")
+  @ApiOperation(value = "rename method configuration in a workspace",
+    nickname = "renamemethodconfig",
+    httpMethod = "Post")
+  @ApiImplicitParams(Array(
+    new ApiImplicitParam(name = "workspaceNamespace", required = true, dataType = "string", paramType = "path", value = "Workspace Namespace"),
+    new ApiImplicitParam(name = "workspaceName", required = true, dataType = "string", paramType = "path", value = "Workspace Name"),
+    new ApiImplicitParam(name = "methodConfigurationNamespace", required = true, dataType = "string", paramType = "path", value = "Method Configuration Namespace"),
+    new ApiImplicitParam(name = "methodConfigurationName", required = true, dataType = "string", paramType = "path", value = "Method Configuration Name"),
+    new ApiImplicitParam(name = "newMethodConfigurationName", required = true, dataType = "string", paramType = "path", value = "New Method Configuration Name")
```
Comment (jmthibault79) [inline src/main/scala/org/broadinstitute/dsde/rawls/webservice/MethodConfigApiService.scala]: This is a `body` parameter of type `org.broadinstitute.dsde.rawls.model.MethodConfigurationName`

File: src/main/scala/org/broadinstitute/dsde/rawls/webservice/MethodConfigApiService.scala [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+  @ApiOperation(value = "Update method configuration in a workspace",
+    nickname = "update method configuration",
+    httpMethod = "Put",
+    produces = "application/json",
+    response = classOf[MethodConfiguration])
+  @ApiImplicitParams(Array(
+    new ApiImplicitParam(name = "workspaceNamespace", required = true, dataType = "string", paramType = "path", value = "Workspace Namespace"),
+    new ApiImplicitParam(name = "workspaceName", required = true, dataType = "string", paramType = "path", value = "Workspace Name"),
+    new ApiImplicitParam(name = "newMethodConfigJson", required = true, dataType = "org.broadinstitute.dsde.rawls.model.MethodConfiguration", paramType = "body", value = "New Method Configuration contents")
+  ))
```
Comment (jmthibault79) [inline src/main/scala/org/broadinstitute/dsde/rawls/webservice/MethodConfigApiService.scala]: Missing `methodConfigurationNamespace` and `methodConfigName`

File: src/main/scala/org/broadinstitute/dsde/rawls/webservice/MethodConfigApiService.scala [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+  @ApiResponses(Array(
+    new ApiResponse(code = 200, message = "Successful Request"),
+    new ApiResponse(code = 404, message = "Workspace or method configuration does not exists"),
+    new ApiResponse(code = 500, message = "Rawls Internal Error")
+  ))
+  def updateMethodConfigurationRoute = cookie("iPlanetDirectoryPro") { securityTokenCookie =>
+    path("workspaces" / Segment / Segment / "methodconfigs" / Segment / Segment ) { (workspaceNamespace, workspaceName, methodConfigurationNamespace, methodConfigName) =>
+      put {
```
Comment (jmthibault79) [inline src/main/scala/org/broadinstitute/dsde/rawls/webservice/MethodConfigApiService.scala]: Update routes should be PATCH

File: src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala [hunk 7]
```diff
@@ -62,14 +66,16 @@ class WorkspaceService(dataSource: DataSource, workspaceDAO: WorkspaceDAO, entit
...
     case GetMethodConfiguration(workspaceNamespace, workspaceName, methodConfigurationNamespace, methodConfigurationName) => context.parent ! getMethodConfiguration(workspaceNamespace, workspaceName, methodConfigurationNamespace, methodConfigurationName)
-    case UpdateMethodConfiguration(workspaceNamespace: String, workspaceName: String, methodConfiguration: MethodConfiguration) => context.parent ! updateMethodConfiguration(workspaceNamespace, workspaceName, methodConfiguration)
-    case CopyMethodConfiguration(workspaceNamespace: String, workspaceName: String, sourceMethodConfigName: MethodConfigurationName) => context.parent ! copyMethodConfiguration(workspaceNamespace, workspaceName, sourceMethodConfigName)
-    case ListMethodConfigurations(workspaceNamespace: String, workspaceName: String) => context.parent ! listMethodConfigurations(workspaceNamespace, workspaceName)
+    case UpdateMethodConfiguration(workspaceNamespace, workspaceName, methodConfiguration: MethodConfiguration) => context.parent ! updateMethodConfiguration(workspaceNamespace, workspaceName, methodConfiguration)
```
Comment (jmthibault79) [inline src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala]: Why remove the `String` type notation but not `MethodConfiguration` here?

**Comment by jmthibault79** [comment]
Done with review.

File: src/main/scala/org/broadinstitute/dsde/rawls/webservice/MethodConfigApiService.scala (hunk 6 above)
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/rawls/webservice/MethodConfigApiService.scala]: but this is not a partial, this is complete overwrite so it should be put

File: src/main/scala/org/broadinstitute/dsde/rawls/webservice/MethodConfigApiService.scala [hunk 8]
```diff
@@ -0,0 +1,213 @@
...
+  @Path("/{configNamespace}/{configName}/rename")
+  @ApiOperation(value = "rename method configuration in a workspace",
+    nickname = "renamemethodconfig",
+    httpMethod = "Post")
+  @ApiImplicitParams(Array(
+    new ApiImplicitParam(name = "workspaceNamespace", required = true, dataType = "string", paramType = "path", value = "Workspace Namespace"),
+    new ApiImplicitParam(name = "workspaceName", required = true, dataType = "string", paramType = "path", value = "Workspace Name"),
+    new ApiImplicitParam(name = "methodConfigurationNamespace", required = true, dataType = "string", paramType = "path", value = "Method Configuration Namespace"),
+    new ApiImplicitParam(name = "methodConfigurationName", required = true, dataType = "string", paramType = "path", value = "Method Configuration Name"),
+    new ApiImplicitParam(name = "newMethodConfigurationName", required = true, dataType = "string", paramType = "body", value = "New Method Configuration Name")
```
Comment (jmthibault79) [inline src/main/scala/org/broadinstitute/dsde/rawls/webservice/MethodConfigApiService.scala]: Update dataType

File: src/main/scala/org/broadinstitute/dsde/rawls/webservice/MethodConfigApiService.scala (hunk 6 above)
Comment (jmthibault79) [inline src/main/scala/org/broadinstitute/dsde/rawls/webservice/MethodConfigApiService.scala]: OK

**Pull request merged**